src/popup/modules/UserInterface.js

```diff
@@ -435,6 +437,37 @@ function triggerFileSave(file, filename, requestDownloadPermissions) {
     });
 }
 
+/**
+ * If qrcode text was an URL i.e contains :// this function generates a filename using the domain and replacing all special characters with _
+ *.If not return "qrcode"
+ *
+ * @function
+ * @private
+ * @param { } 
+ * @returns {filename}
+ */
+function generateFilename(){
+	var firstIndex = qrCodeInputText.indexOf("://");
+	if(firstIndex > 0) {
+		// URL found
+		var filename = qrCodeInputText.substring(firstIndex+3);
+		var wwwIndex = filename.indexOf("www");
+		if(wwwIndex >= 0){
+			// remove www(*.).
+			filename = filename.substring(filename.indexOf(".")+1);
+		}
+		// keep text until first "/"
+		filename = filename.substring(0,filename.indexOf("/"));
+		// replace . by _
+		//filename = filename.replace(/\./g,"_");
```

Comment:
so why is this uncommented? Shall we include it now or not?

IMHO at least Windows is quite sensitive with too many dots in your filename. So IMHO replace it. (But line 463 should also already do that, should not it?)

Reply:
It is replaced by the replace()